Assets/GRPC.NET/Scripts/GRPCBestHttpHandler.cs

```diff
@@ -144,20 +131,68 @@ protected override Task<HttpResponseMessage> SendAsync(HttpRequestMessage grpcRe
                 isHeader = false;
             };
 
-
-            // For each incoming DATA package we write data trough to gRPC
-            bestRequest.OnStreamingData += (_, _, fragment, length) =>
+            // For each incoming DATA package we write data through to gRPC
+            bestRequest.DownloadSettings.OnDownloadStarted += (_, _, stream) =>
             {
-                // Write incoming DATA package and immediately flush
-                incomingDataStream.Write(fragment, 0, length);
-                incomingDataStream.Flush();
-                return true;
+                try
+                {
+                    // Copy what we have already available, we'll download the rest
+                    // async if needed in a streaming mode detached from the response.
+                    while (stream.TryTake(out var buffer))
+                    {
+                        // Make sure that the buffer is released back to the BufferPool.
+                        using var b = buffer.AsAutoRelease();
+                        incomingDataStream.Write(b.Data, b.Offset, b.Count);
+                    }
+
+                    // That's all the data we have, so we're done streaming.
+                    if (stream.IsCompleted)
+                    {
+                        incomingDataStream.Close();
+                        return;
+                    }
+                }
+                catch (Exception e)
+                {
+                    incomingDataStream.CloseWithException(e);
+                    return;
+                }
+
+                // Not all the data was available right away, so lets detach
+                // the stream and consume the rest async in a Task.
+                stream.IsDetached = true;
+                Task.Run(async () =>
+                {
+                    try
+                    {
+                        while (!stream.IsCompleted)
+                        {
+                            if (stream.TryTake(out var buffer))
+                            {
+                                // Make sure that the buffer is released back to the BufferPool.
+                                using var b = buffer.AsAutoRelease();
+                                incomingDataStream.Write(b.Data, b.Offset, b.Count);
+                            }
+                            else
+                            {
+                                // The DownloadContentStream does not block, so wait a bit.
+                                await Task.Delay(ASYNC_READ_WAIT_MS, cancellationToken);
```

Comment:
never happy about polling delays, but it is what it is. havent seen any other available notice/signal pattern in Best HTTP, but I may have missed it.